packages/pwa-kit-build/src/configs/webpack/config.js

```diff
@@ -8,325 +8,14 @@
 /* eslint-env node */
 
 // For more information on these settings, see https://webpack.js.org/configuration
-import fs from 'fs'
-import path, {resolve} from 'path'
-
-import webpack from 'webpack'
-import WebpackNotifierPlugin from 'webpack-notifier'
-import CopyPlugin from 'copy-webpack-plugin'
-import {BundleAnalyzerPlugin} from 'webpack-bundle-analyzer'
-import LoadablePlugin from '@loadable/webpack-plugin'
-import SpeedMeasurePlugin from 'speed-measure-webpack-plugin'
-import {createModuleReplacementPlugin, PwaKitConfigPlugin} from './plugins'
+const fs = require('fs')
+const path = require('path')
 
 const projectDir = process.cwd()
-const sdkDir = path.resolve(path.join(__dirname, '..', '..', '..'))
-
-const pkg = require(resolve(projectDir, 'package.json'))
-const buildDir = resolve(projectDir, 'build')
-
-const production = 'production'
-const development = 'development'
-const analyzeBundle = process.env.MOBIFY_ANALYZE === 'true'
-const mode = process.env.NODE_ENV === production ? production : development
-const DEBUG = mode !== production && process.env.DEBUG === 'true'
-const CI = process.env.CI
-
-if ([production, development].indexOf(mode) < 0) {
-    throw new Error(`Invalid mode "${mode}"`)
-}
-
-const entryPointExists = (segments) => {
-    for (let ext of ['.js', '.jsx', '.ts', '.tsx']) {
-        const p = path.resolve(projectDir, ...segments) + ext
-        if (fs.existsSync(p)) {
-            return true
-        }
-    }
-    return false
-}
+const projectWebpackPath = path.resolve(projectDir, 'webpack.config.js')
 
-const findInProjectThenSDK = (pkg) => {
-    const projectPath = resolve(projectDir, 'node_modules', pkg)
-    return fs.existsSync(projectPath) ? projectPath : resolve(sdkDir, 'node_modules', pkg)
+if (fs.existsSync(projectWebpackPath)) {
+    module.exports = require(projectWebpackPath)
+} else {
+    module.exports = require('./base-config.js')
```

Comment:
Suppose users completely replace the Webpack configuration we provide. In that case, they will need to copy and paste all the Webpack configuration that we require into the project to get things working again and then customize it. 

Once they choose to use their own Webpack configuration file in the project, users will completely control the Webpack configuration. But, at the same time, users will no longer benefit from the updates that we provide to our Webpack configuration.

In my opinion, this is not the best API experience to customize the Webpack configuration. I think the experience provided in V1 has some advantages that we should consider.

V1 allows users to override parts of the configuration or extend the Webpack configuration. This approach ensures that users consume future updates and that the configuration is less likely to break. It has the disadvantage that they need to use our Webpack configuration.

@vcua-mobify Are you open to summarizing the possible approaches in a Technical proposal doc and discussing it with the team during our Technical discussion meeting before deciding how we want to allow users to customize the Webpack configuration in V2?

Reply:
I'd have to agree with @adamraya on this. You approach almost seems like a partial implementation of `create-react-app`'s eject feature, minus we aren't providing the automation of ejecting. I can't say I love either our approach or the new one, but the existing V1 behaviour is the lesser of 2 evils in my eyes.